test/service/test_manifest.py

```diff
@@ -440,14 +440,42 @@ def test_manifest_zarr(self):
                 self.assertEqual(200, response.status_code)
                 lines = response.content.decode().splitlines()
                 file_prefix = 'output="587d74b4-1075-4bbf-b96a-4d1ede0481b2/'
+                location_prefix = f'url="{config.service_endpoint()}/repository/files'
                 curl_files = []
+                curl_download_urls = []
+                curl_aux_download_urls = []
                 for line in lines:
                     if line.startswith(file_prefix):
                         self.assertTrue(line.endswith('"'))
                         file_name = line[len(file_prefix):-1]
                         curl_files.append(file_name)
+                    elif line.startswith(location_prefix):
+                        self.assertTrue(line.endswith('"'))
+                        download_url = line[len(location_prefix):-1]
+                        url = furl(download_url)
+                        try:
+                            url.args['drsPath']
+                        except KeyError:
+                            curl_download_urls.append(url)
+                        else:
+                            curl_aux_download_urls.append(url)
+                    else:
+                        # The curl manifest contains a combination of curl
+                        # commands, blank lines, `output` and `url` destinations.
+                        # To verify correctness we only check `output` and `url`,
+                        # everything else can be disregarded.
+                        pass
                 self.assertEqual(sorted([f['file_name'] for f in expected]),
                                  sorted(curl_files))
+                # Related (aux) files from zarr store are not indexed in Azul as
+                # other files are, so to ensure that the curl manifest download
+                # provisions the files correctly, their location url contains
+                # `drsPath`, `fileName` and `requestIndex` as query parameters …
+                for url in curl_aux_download_urls:
```

Comment:
The usual problem with your test approach. If the `url=` line were missing for related files, or had the wrong prefix, this would still pass.